src/kritter/ktextvisor.py

```diff
@@ -0,0 +1,298 @@
+        except:
+            return []
+
+    def post_handle_context(self, sender, responses):
+        t = time.time()
+        contexts = [r.context for r in responses]
+        contexts = [c for c in contexts if c is not None]
+        if len(contexts):
+            context = set(contexts[-1])
+        else:
+            try:
+                context = self.context[sender['id']][1]
+            except:
+                context = []
+        if context==[]:
+            try:
+                del self.context[sender['id']]
+            except: 
+                pass
+        else:
+            self.context[sender['id']] = (t, context)
+
+                        
+    def send_responses(self, sender, responses):
+        content = [r.content for r in responses] # extract content
+        content = format_content(content) # combine strs and dicts
+        content = [format_dict(c) if isinstance(c, dict) else c for c in content] # format dicts into strs
+        content = format_content(content) # combine strs again
+        with self.lock:
+            self.text_client.send(content, sender)
+
+    def native_callback(self, words, sender, context):
+        # userids need to be saved as strings to avoid
+        # duplication error related to json.loads()
+        # duplication appears as { 1: 'name': '1': 'name' }
+        def subscribe(words, sender, context):
+            output = "error subscribing"
+            self.config.load()
+            subscribers = self.config['subscribers']
+            userid = str(sender['id'])
+            if not userid in subscribers.keys():
+                subscribers[userid] = sender['name']
+                output = f"{sender['name']} is now subscribed"
+            else:
+                output = f"{sender['name']} is already subscribed"
+            self.config['subscribers'] = subscribers
+            self.config.save()
+            return output
+
+        def unsubscribe(words, sender, context):
+            output = "error unsubscribing"
+            self.config.load()
+            subscribers = self.config['subscribers']
+            userid = str(sender['id'])
+            if userid in subscribers.keys():
+                del subscribers[userid]
+                output = f"{sender['name']} has been unsubscribed"
+            else:
+                output = f"{sender['name']} was not subscribed"
+            self.config['subscribers'] = subscribers
+            self.config.save()
+            return output
+
+        tv_table = KtextVisorTable({
+            "subscribe": (subscribe, "Subscribe self to Vizy updates."),
+            "unsubscribe": (unsubscribe, "Unsubscribe self from Vizy updates.")})
+        @self.callback_receive()
+        def func(words, sender, context):
+            return tv_table.lookup(words, sender, context)            
+
+    def send(self, msg, to):
+        if to is None:
+            # send to all subscribers using their Ids
+            self.config.load()
```

Comment:
I don't think you need to call config.load().  You load it in __init__ (why would it change?)